app/factories/flood_risk_engine/steps/form_object_factory.rb

```diff
@@ -19,23 +19,15 @@ def form_object_class_map
             check_exemptions:                 Steps::NullForm,
             user_type:                        Steps::UserTypeForm,
             check_your_answers:               Steps::CheckYourAnswersForm,
-            declaration:                      Steps::NullForm,
-            partnership_details:              Steps::NullForm
+            declaration:                      Steps::NullForm
           }
         end
 
         private
 
         def setup_form_object(step)
           form_name = "FloodRiskEngine::Steps::#{step.to_s.classify}Form"
-          begin
-            form_name.constantize
-          rescue NameError => x
-            Rails.logger.debug(x.backtrace.first)
-            Rails.logger.debug(x.inspect)
-            Rails.logger.debug("Error loading Form class #{form_name} ")
-            nil
-          end
+          form_name.constantize
```

Comment:
Because the default error raised when the name can't be converted into a constant is more informative than the custom error. That is, you get an error that tells you precisely what the problem is.

When I started building the form, I gave it the wrong name. I couldn't work out what the problem was until I removed the custom error catching. As the default error was more helpful in debugging the problem, I didn't put the customer error catch back in.